### PR TITLE
feat(textures): basis - auto-select decoder format

### PR DESCRIPTION
**Pull request by belom88**

### CODE REVIEW

File: modules/textures/src/lib/parsers/parse-basis.js [hunk 1]
```diff
@@ -29,6 +30,14 @@ const OutputFormat = {
  * @returns compressed texture data
  */
 export default async function parseBasis(data, options) {
+  if (options.basis.decoderFormat === 'auto') {
+    if (isKTX(data)) {
+      const fileConstructors = await loadBasisEncoderModule(options);
+      return parseKTX2File(fileConstructors.KTX2File, data, options);
```
Comment (ibgreen) [inline modules/textures/src/lib/parsers/parse-basis.js:36]: Does this handle the case when the file is basis without KTX? Can you add some comments and JSDoc?

Side note: I wonder if we would be able to create a `KTX2Loader`?

File: modules/textures/test/basis-loader.spec.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ import {load, setLoaderOptions, isBrowser} from '@loaders.gl/core';
 import {GL} from '../src/lib/gl-constants';
 
 const TEST_URL = '@loaders.gl/textures/test/data/alpha3.basis';
```
Comment (ibgreen) [inline modules/textures/test/basis-loader.spec.js:7]:
```suggestion
const BASIS_TEST_URL = '@loaders.gl/textures/test/data/alpha3.basis';
```

File: modules/textures/test/basis-loader.spec.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import {load, setLoaderOptions, isBrowser} from '@loaders.gl/core';
...
+const KTX2_TEST_URL = '@loaders.gl/textures/test/data/kodim23.ktx2';
```
Comment (ibgreen) [inline modules/textures/test/basis-loader.spec.js:8]:
```suggestion
const KTX2_BASIS_TEST_URL = '@loaders.gl/textures/test/data/kodim23.ktx2';
```

File: modules/textures/test/basis-loader.spec.js [hunk 4]
```diff
@@ -105,6 +106,30 @@ test('BasisLoader#transcode to explicit format', async (t) => {
   t.end();
 });
 
+test('BasisLoader#auto-select a decoder format', async (t) => {
+  const images = await load(TEST_URL, BasisLoader, {
+    worker: true,
+    basis: {
+      format: 'astc-4x4',
+      decoderFormat: 'auto'
```
Comment (ibgreen) [inline modules/textures/test/basis-loader.spec.js:114]: Could this be `containerFormat`? (In the sense that we are dealing with basis encoded data inside a KTX 2 "container"?

**Review by ibgreen — COMMENTED** [review]
Let's not forget to update docs.

**Review by ibgreen — APPROVED** [review]

File: modules/textures/src/lib/parsers/parse-basis.js (hunk 1 above)
Comment (belom88) [reply modules/textures/src/lib/parsers/parse-basis.js:36]: It does

File: modules/textures/test/basis-loader.spec.js (hunk 2 above)
Comment (belom88) [reply modules/textures/test/basis-loader.spec.js]: Done

File: modules/textures/test/basis-loader.spec.js (hunk 3 above)
Comment (belom88) [reply modules/textures/test/basis-loader.spec.js]: Done

File: modules/textures/test/basis-loader.spec.js (hunk 4 above)
Comment (belom88) [reply modules/textures/test/basis-loader.spec.js]: Done

**Pull request merged**